File: Stack_Queue/18258.py
import sys
sinput = sys.stdin.readline

queue = list()
num = int(sinput())
cnt = 0
for _ in range(num):
    inp = list(sinput().split())

    if inp[0] == "push":
        queue.append(int(inp[1]))
    elif inp[0] == "pop":
        if len(queue)-cnt == 0:
            print(-1)
        else:
            print(queue[cnt])
            cnt += 1
    elif inp[0] == "size":
        print(len(queue)-cnt)
    elif inp[0] == "empty":
        if len(queue)-cnt == 0:
            print(1)
        else:
            print(0)
    elif inp[0] == "front":
        if len(queue)-cnt != 0:
            print(queue[cnt])
        else:
            print(-1)
    elif inp[0] == "back":
        if len(queue)-cnt != 0:
            print(queue[-1])
        else:
            print(-1)

File: barkingDog series/0x06_queue/18258.cpp
#include <iostream>
#include <queue>
#include <string>

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    int n;
    cin >> n;

    queue<int> q;

    while (n--) {
        string inp;
        cin >> inp;
        
        if (inp == "push") {
            int k;
            cin >> k;
            q.push(k);
        } else if (inp == "pop") {
            if (q.empty()) {
                cout << -1 << "\n";
                continue;
            }
            cout << q.front() << "\n";
            q.pop();
        } else if (inp == "size") {
            cout << q.size() << "\n";
        } else if (inp == "empty") {
            cout << q.empty() << "\n";
        } else if (inp == "front") {
            if (q.empty()) {
                cout << -1 << "\n";
                continue;
            }
            cout << q.front() << "\n";
        } else {
            if (q.empty()) {
                cout << -1 << "\n";
                continue;
            }
            cout << q.back() << "\n";
        }
    }
}